core/src/main/scala/kafka/consumer/TopicCount.scala

```diff
@@ -104,6 +104,9 @@ private[kafka] object TopicCount extends Logging {
   def constructTopicCount(consumerIdString: String, filter: TopicFilter, numStreams: Int, zkUtils: ZkUtils, excludeInternalTopics: Boolean) =
     new WildcardTopicCount(zkUtils, consumerIdString, filter, numStreams, excludeInternalTopics)
 
+  override def equals(o: Any): Boolean = {
+    throw new KafkaException("can't use equals here")
+  }
```

Comment:
`WildcardTopicCount` does not override `equals` method at all. Generally, it is quite bad idea to throw exception during object comparison. Rather, I would get rid of `StaticTopicCount#equals` by making it case class.